src/dialect/postgresql.rs

```diff
@@ -42,6 +42,10 @@ impl Dialect for PostgreSqlDialect {
             None
         }
     }
+
+    fn supports_filter_during_aggregation(&self) -> bool {
```

Comment:
For what it is worth, I think the DataFusion actually uses the Generic Dialect, https://github.com/apache/arrow-datafusion/blob/0084aeb686b318cbdb49cab00cb8f15c9f520d1e/datafusion/sql/src/parser.rs#L102

Thus we may want to add return true from `supports_filter_during_aggregation` in https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/dialect/generic.rs as well

Reply:
Thanks. Dask SQL has their own dialect where they want to use this, but it makes sense to add to the dialect that DataFusion is using as well.